core/templates/domain/topics_and_skills_dashboard/topics-and-skills-dashboard-backend-api.service.spec.ts

```diff
@@ -13,98 +13,83 @@
-  beforeEach(angular.mock.module('oppia', function($provide) {
-    var ugs = new UpgradedServices();
-    for (let [key, value] of Object.entries(ugs.getUpgradedServices())) {
-      $provide.value(key, value);
-    }
-  }));
-
-  beforeEach(angular.mock.inject(function($injector) {
-    TopicsAndSkillsDashboardBackendApiService = $injector.get(
-      'TopicsAndSkillsDashboardBackendApiService');
-    UrlInterpolationService = $injector.get('UrlInterpolationService');
-    $httpBackend = $injector.get('$httpBackend');
-  }));
 
-  afterEach(function() {
-    $httpBackend.verifyNoOutstandingExpectation();
-    $httpBackend.verifyNoOutstandingRequest();
-  });
-
-  it('should successfully fetch topics and skills dashboard data from the ' +
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+    });
+    httpTestingController = TestBed.get(HttpTestingController);
+    topicsAndSkillsDashboardBackendApiService = TestBed.get(
+      TopicsAndSkillsDashboardBackendApiService);
+    let sampleDataResults = {
+      topic_summary_dicts: [{
+        id: SAMPLE_TOPIC_ID,
+        name: 'Sample Name',
+        language_code: 'en',
+        version: 1,
+        canonical_story_count: 3,
+        additional_story_count: 0,
+        uncategorized_skill_count: 3,
+        subtopic_count: 3,
+        topic_model_created_on: 1466178691847.67,
+        topic_model_last_updated: 1466178759209.839
+      }],
+      skill_summary_dicts: []
+    };
+
+    afterEach(() => {
+      httpTestingController.verify();
+    });
+
+    it('should use rejection handler if dashboard data backend request failed',
+      fakeAsync(() => {
+        var successHandler = jasmine.createSpy('success');
+        var failHandler = jasmine.createSpy('fail');
+        topicsAndSkillsDashboardBackendApiService.fetchDashboardData().then(
+          successHandler, failHandler);
+        var req = httpTestingController.expectOne(
+          TOPICS_AND_SKILLS_DASHBOARD_DATA_URL,
+          'Error Loading dashboard data.');
+        expect(req.request.method).toEqual('GET');
+        expect(successHandler).not.toHaveBeenCalled();
+        expect(failHandler).toHaveBeenCalled();
+      })
+    );
+
+    it('should successfully fetch topics and skills dashboard data from the ' +
       'backend',
-  function() {
-    var successHandler = jasmine.createSpy('success');
-    var failHandler = jasmine.createSpy('fail');
-
-    $httpBackend.expect('GET', TOPICS_AND_SKILLS_DASHBOARD_DATA_URL).respond(
-      sampleDataResults);
-    TopicsAndSkillsDashboardBackendApiService.fetchDashboardData().then(
-      successHandler, failHandler);
-    $httpBackend.flush();
-
-    expect(successHandler).toHaveBeenCalled();
-    expect(failHandler).not.toHaveBeenCalled();
-  });
-
-  it('should use rejection handler if dashboard data backend request failed',
-    function() {
+    fakeAsync(() => {
```

Comment:
```suggestion
    it('should successfully fetch topics and skills dashboard data from the ' +
      'backend', fakeAsync(() => {
```

Reply:
Here and elsewhere.

Reply:
fakeAsync() should be in the same line as that of the backend ?